src/com/lld/inmemorycache/model/impl/LRUEvictionPolicy.java

```diff
@@ -0,0 +1,70 @@
+package com.lld.inmemorycache.model.impl;
+
+import com.lld.inmemorycache.model.DoublyLinkedList;
+import com.lld.inmemorycache.model.IEvictionPolicy;
+import com.lld.inmemorycache.model.Node;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class LRUEvictionPolicy implements IEvictionPolicy {
+    private DoublyLinkedList keys;
+    private HashMap<String, Node> mapper;
+
+    private ReentrantLock lock;
+
+    public LRUEvictionPolicy() {
+        keys = new DoublyLinkedList();
+        mapper = new LinkedHashMap<>();
+        lock = new ReentrantLock(true);
+    }
+
+    @Override
+    public void keyAccessed(String key) {
+        lock.lock();

```

Comment:
As mentioned, taking individual locks at the storage or eviction layer won't work.